lib/urlUtil.js

```diff
@@ -20,6 +20,19 @@ const defaultToHttpScheme = (url) => {
   return url
 }
 
+// https://nodejs.org/api/url.html#url_url_strings_and_url_objects
+// contains a good graphical representation of the URL specification
+const getBaseUrl = () => `${window.location.protocol}//${window.location.host}`
```

Comment:
these extra Utils not required as functionality is in the Next Router.
import { useRouter } from 'next/router'
const router = useRouter()
{router.asPath}. instead of getPathAndHash

Reply:
useRouter hook only works inside the body of a functional react component.

Just tried it and get the following error:

```
  Error {
    message: `Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:␊
    1. You might have mismatching versions of React and the renderer (such as React DOM)␊
    2. You might be breaking the Rules of Hooks␊
    3. You might have more than one copy of React in the same app␊
    See https://fb.me/react-invalid-hook-call for tips about how to debug and fix this problem.`,
  }
```

Reply:
ah ok.  I'm sure there's a server side version but I'll pull this in for now.